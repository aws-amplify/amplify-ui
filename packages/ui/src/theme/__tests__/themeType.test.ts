import { Theme } from '../index';

describe('Amplify UI Theme', () => {
  describe('components', () => {
    it('should allow empty components', () => {
      const theme: Theme = {
        name: 'test',
        tokens: {
          components: {
            alert: {},
            authenticator: {},
            autocomplete: {},
            badge: {},
            button: {},
            card: {},
            checkbox: {},
            checkboxfield: {},
            collection: {},
            copy: {},
            divider: {},
            expander: {},
            field: {},
            fieldcontrol: {},
            fieldgroup: {},
            fieldmessages: {},
            fileuploader: {},
            flex: {},
            heading: {},
            highlightmatch: {},
            icon: {},
            image: {},
            inappmessaging: {},
            link: {},
            loader: {},
            menu: {},
            pagination: {},
            passwordfield: {},
            phonenumberfield: {},
            placeholder: {},
            radio: {},
            radiogroup: {},
            rating: {},
            searchfield: {},
            select: {},
            selectfield: {},
            sliderfield: {},
            stepperfield: {},
            switchfield: {},
            table: {},
            tabs: {},
            text: {},
            textareafield: {},
            textfield: {},
            togglebutton: {},
            togglebuttongroup: {},
          },
        },
      };
      // This test doesn't test anything, but
      // if there is a TS error it will fail
      expect(theme).toMatchInlineSnapshot(`
        Object {
          "name": "test",
          "tokens": Object {
            "components": Object {
              "alert": Object {},
              "authenticator": Object {},
              "autocomplete": Object {},
              "badge": Object {},
              "button": Object {},
              "card": Object {},
              "checkbox": Object {},
              "checkboxfield": Object {},
              "collection": Object {},
              "copy": Object {},
              "divider": Object {},
              "expander": Object {},
              "field": Object {},
              "fieldcontrol": Object {},
              "fieldgroup": Object {},
              "fieldmessages": Object {},
              "fileuploader": Object {},
              "flex": Object {},
              "heading": Object {},
              "highlightmatch": Object {},
              "icon": Object {},
              "image": Object {},
              "inappmessaging": Object {},
              "link": Object {},
              "loader": Object {},
              "menu": Object {},
              "pagination": Object {},
              "passwordfield": Object {},
              "phonenumberfield": Object {},
              "placeholder": Object {},
              "radio": Object {},
              "radiogroup": Object {},
              "rating": Object {},
              "searchfield": Object {},
              "select": Object {},
              "selectfield": Object {},
              "sliderfield": Object {},
              "stepperfield": Object {},
              "switchfield": Object {},
              "table": Object {},
              "tabs": Object {},
              "text": Object {},
              "textareafield": Object {},
              "textfield": Object {},
              "togglebutton": Object {},
              "togglebuttongroup": Object {},
            },
          },
        }
      `);
    });
  });
});
