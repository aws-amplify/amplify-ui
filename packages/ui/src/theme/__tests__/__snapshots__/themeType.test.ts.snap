// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amplify UI Theme components should allow empty components 1`] = `
{
  "name": "test",
  "tokens": {
    "components": {
      "accordion": {},
      "alert": {},
      "authenticator": {},
      "autocomplete": {},
      "badge": {},
      "button": {},
      "card": {},
      "checkbox": {},
      "checkboxfield": {},
      "collection": {},
      "copy": {},
      "divider": {},
      "field": {},
      "fieldcontrol": {},
      "fieldgroup": {},
      "fieldmessages": {},
      "flex": {},
      "heading": {},
      "highlightmatch": {},
      "icon": {},
      "image": {},
      "inappmessaging": {},
      "link": {},
      "liveness": {},
      "loader": {},
      "menu": {},
      "pagination": {},
      "passwordfield": {},
      "phonenumberfield": {},
      "placeholder": {},
      "radio": {},
      "radiogroup": {},
      "rating": {},
      "searchfield": {},
      "select": {},
      "selectfield": {},
      "sliderfield": {},
      "stepperfield": {},
      "switchfield": {},
      "table": {},
      "tabs": {},
      "text": {},
      "textareafield": {},
      "textfield": {},
      "togglebutton": {},
      "togglebuttongroup": {},
    },
  },
}
`;
